pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -387,16 +431,16 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
                         limitRaw: limits[i]
                     })
                 );
-            } else {
-                underlyingAmounts[i] = 0;
-                wrappedAmounts[i] = 0;
             }
 
             if (isStaticCall == false && kind == SwapKind.EXACT_OUT) {
+                bool wethIsEth = params.wethIsEth;
                 // If the SwapKind is EXACT_OUT, the limit of underlying tokens was taken from the user, so the
                 // difference between limit and exact underlying amount needs to be returned to the sender.
-                _sendTokenOut(params.sender, underlyingToken, limits[i] - underlyingAmounts[i], params.wethIsEth);
+                _sendTokenOut(params.sender, underlyingToken, limits[i] - underlyingAmount, wethIsEth);
             }
+
+            finalAmountsIn[i] = kind == SwapKind.EXACT_IN ? wrappedAmount : underlyingAmount;
```

Comment:
I don't understand why this is like this.

This function always wraps, so isn't `tokenIn` always underlying if we get to this point? Am I missing something?